src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -309,10 +309,9 @@ public void WriteTo(TextWriter writer, HtmlEncoder encoder)
                         using (var stringWriter = new StringWriter())
                         {
                             htmlContent.WriteTo(stringWriter, encoder);
+                            stringWriter.GetStringBuilder().Replace("\"", "&quot;");
```

Comment:
Doing this replace on the `StringBuilder`/`char[]` before creating a string is better.